examples/map/map_portable_versioning_example.py

```diff
@@ -0,0 +1,186 @@
+
+    def write_portable(self, writer):
+        writer.write_utf("name", self.name)
+        writer.write_utf("age", self.age)
+        writer.write_utf("manager", self.manager)
+
+    def read_portable(self, reader):
+        self.name = reader.read_utf("name")
+        self.age = reader.read_utf("age")
+        self.manager = reader.read_utf("manager")
+
+    def get_factory_id(self):
+        return self.FACTORY_ID
+
+    def get_class_id(self):
+        return self.CLASS_ID
+
+    def get_class_version(self):
+        return self.CLASS_VERSION
+
+    def __str__(self):
+        return "Employee[ name:{} age:{} manager:{} ]".format(self.name, self.age, self.manager)
+
+    def __eq__(self, other):
+        return type(self) == type(other) and self.name == other.name and self.age == other.age \
+               and self.manager == other.manager
+
+
+if __name__ == '__main__':
+
+    # Let's now configure 3 clients with 3 different versions of Employee.
+    config = hazelcast.ClientConfig()
+    config.serialization_config.portable_factories[Employee.FACTORY_ID] = \
+        {Employee.CLASS_ID: Employee}
+    client = hazelcast.HazelcastClient(config)
+
+    config2 = hazelcast.ClientConfig()
+    config2.serialization_config.portable_factories[Employee2.FACTORY_ID] = \
+        {Employee2.CLASS_ID: Employee2}
+    client2 = hazelcast.HazelcastClient(config2)
+
+    config3 = hazelcast.ClientConfig()
+    config3.serialization_config.portable_factories[Employee3.FACTORY_ID] = \
+        {Employee3.CLASS_ID: Employee3}
+    client3 = hazelcast.HazelcastClient(config3)
+
+    # Assume that a member joins a cluster with a newer version of a class.
+    # If you modified the class by adding a new field, the new member's put operations include that
+    # new field.
+    my_map = client.get_map("employee-map").blocking()
+    my_map2 = client2.get_map("employee-map").blocking()
+
+    my_map.clear()
+    my_map.put(0, Employee("Jack", 28))
+    my_map2.put(1, Employee2("Jane", 29, "Josh"))
+
+    print('Map Size: {}'.format(my_map.size()))
+
+    # If this new member tries to get an object that was put from the older members, it
+    # gets null for the newly added field.
+    for v in my_map.values():
+        print(v)
+
+    for v in my_map2.values():
+        print(v)
+
+    # Let's try now to put a version 3 Employee object to the map and see what happens.
+    my_map3 = client3.get_map("employee-map").blocking()
+    my_map3.put(2, Employee3("Joe", "30", "Mary"))
+
+    print('Map Size: {}'.format(my_map.size()))
+
+    # As clients with incompatible versions of the class try to access each other, a HazelcastSerializationError
+    # is raised (caused by a TypeError).
+    try:
+        # Client that has class with int type age field tries to read Employee3 object with String age field.
+        print(my_map.get(2))
+    except hazelcast.exception.HazelcastSerializationError as ex:
+        print("Failed due to: {}".format(ex.message))
```

Comment:
`ex.message` statement throws error since `HazelcastSerializationError`does not have that field.  Just change it to `print("Failed due to: {}".format(ex))` to print the cause.